### PR TITLE
Update virtual lidar to use coordinate transformations and add period…

### PR DESCRIPTION
**Pull request by tonyinme**
This adds the ability to have a scan pattern that repeats itself in time.

The computation on the virtual lidar locations has been changed from vector transformations to a coordinate system transformation. This computation is more general, fixes a bug and simplifies the computation.

### CODE REVIEW

**Comment by ndevelder** [comment]
Hi Tony! This generally looks good. I might be missing it, or misunderstanding how it works, but do lines 80-84 in LidarSampler.cpp assume a directionality to the cartesian coordinate system...i.e. the 2 component is the cos of elevation...does this mean we can only use this in cases where "z" is the "up" direction? If so, we may want to consider some user input to define this directionality, or try and intuit it from the boundary conditions?

File: amr-wind/utilities/sampling/LidarSampler.H [hunk 1]
```diff
@@ -46,6 +46,8 @@ protected:
     amrex::Vector<amrex::Real> m_azimuth_table;
     amrex::Vector<amrex::Real> m_elevation_table;
     amrex::Real m_length{0};
+    bool m_periodic{true};
+    amrex::Real m_period{99999999};
```
Comment (michaeljbrazell) [inline amr-wind/utilities/sampling/LidarSampler.H:50]: use a std max numeric limits please

File: amr-wind/utilities/sampling/LidarSampler.cpp [hunk 2]
```diff
@@ -55,32 +65,23 @@ void LidarSampler::update_sampling_locations()
...
     amrex::Real time = m_sim.time().current_time();
 
-    // The current azimuth angle
-    const amrex::Real current_azimuth =
-        ::amr_wind::interp::linear(m_time_table, m_azimuth_table, time);
-
-    const amrex::Real current_elevation =
-        ::amr_wind::interp::linear(m_time_table, m_elevation_table, time);
+    // If the time table is periodic, make sure to get the correct time
+    if (m_periodic) time = std::fmod(time, m_period);
```
Comment (michaeljbrazell) [inline amr-wind/utilities/sampling/LidarSampler.cpp:69]: add braces here please

Comment (tonyinme) [reply amr-wind/utilities/sampling/LidarSampler.cpp]: fixed

File: amr-wind/utilities/sampling/LidarSampler.H (hunk 1 above)
Comment (tonyinme) [reply amr-wind/utilities/sampling/LidarSampler.H]: fixed

**Comment by tonyinme** [comment]
Thanks @michaeljbrazell , the fixes have been implemented.

**Review by michaeljbrazell — APPROVED** [review]

**Pull request merged**